.github/workflows/ci.yml

```diff
@@ -179,31 +186,159 @@ jobs:
+          name: 'terraform-policy-files'
+          path: |
+            ${{ env.POLICY_TF_VAR_FILE }}
+            ${{ env.POLICY_TF_MODULE_DIR }}/terraform.tfstate
+            ${{ env.POLICY_TF_MODULE_DIR }}/terraform.tfstate.backup
+          retention-days: 1
+
+  # Call reuseble workflow
+  call_mapping_reusble_workflow:
+    needs: ['set_up_integration_infra']
+    uses: 'abcxyz/pmap/.github/workflows/snapshot-file-copy.yml@main' # ratchet:exclude
+    with:
+      workload_identity_provider: 'projects/921163060412/locations/global/workloadIdentityPools/github-pool-3b39/providers/github-provider'
+      service_account: 'pmap-3b39-ci-sa@pmap-ci.iam.gserviceaccount.com'
+      destination_prefix: 'pmap-ci/mapping/${{ needs.set_up_integration_infra.outputs.INTEG_TEST_OBJECT_PREFIX_OUTPUT }}'
+      path: 'test/e2e/testdata/fakedata/mapping'
+
+  call_policy_reusble_workflow:
+    needs: ['set_up_integration_infra']
+    uses: 'abcxyz/pmap/.github/workflows/snapshot-file-copy.yml@main' # ratchet:exclude
+    with:
+      workload_identity_provider: 'projects/921163060412/locations/global/workloadIdentityPools/github-pool-3b39/providers/github-provider'
+      service_account: 'pmap-3b39-ci-sa@pmap-ci.iam.gserviceaccount.com'
+      destination_prefix: 'pmap-ci/policy/${{ needs.set_up_integration_infra.outputs.INTEG_TEST_OBJECT_PREFIX_OUTPUT }}'
+      path: 'test/e2e/testdata/fakedata/policy'
+
+  run_integ_test:
+    runs-on: 'ubuntu-latest'
+    needs: ['set_up_integration_infra', 'call_mapping_reusble_workflow', 'call_policy_reusble_workflow']
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    env:
+      TF_MODULE_DIR: './terraform/modules/pmap-service'
+      MAPPING_TF_MODULE_DIR: '/tmp/mapping_tf_module'
+      POLICY_TF_MODULE_DIR: '/tmp/policy_tf_module'
+      INTEG_TEST_PROJECT_ID: 'pmap-ci'
+      INTEG_TEST_BUCKET_ID: 'pmap-ci'
+      INTEG_TEST_BIGQUERY_DATASET_ID: 'pmap_86d8'
+      INTEG_TEST_MAPPING_TABLE_ID: 'mapping'
+      INTEG_TEST_MAPPING_FAILURE_TABLE_ID: 'mapping-failure'
+      INTEG_TEST_POLICY_TABLE_ID: 'policy'
+      INTEG_TEST_MAPPING_DOWNSTREAM_TOPIC: 'mapping-bigquery'
+      INTEG_TEST_STATIC_GCS_BUCKET: 'pmap-static-ci-bucket-7faa'
+    steps:
+      - uses: 'actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c' # ratchet:actions/checkout@v3
+      - id: 'auth'
+        uses: 'google-github-actions/auth@ef5d53e30bbcd8d0836f4288f5e50ff3e086997d' # ratchet:google-github-actions/auth@v1
+        with:
+          workload_identity_provider: '${{ env.WIF_PROVIDER }}'
+          service_account: '${{ env.WIF_SERVICE_ACCOUNT }}'
+          token_format: 'access_token'
+      # Write the prefix into GITHUB_ENV which will be used in integration test as a prefix of the
+      # uploaded file's path.
+      - name: 'Write object prefix'
+        run: |
+          echo "INTEG_TEST_OBJECT_PREFIX=${{ needs.set_up_integration_infra.outputs.INTEG_TEST_OBJECT_PREFIX_OUTPUT }}" >> $GITHUB_ENV
```

Comment:
Why it is written again?

Reply:
Because the first time we write this, it was in a different job, when switch to a new job, the `github_env` doesn't exist anymore. And this value is required in the integ test, the config get this value for env, and set it as the path prefix for uploading yaml files.

Reply:
I wonder if we can set all the env vars at the top level...